ros/src/system/autoware_health_checker/src/diag_buffer.cpp

```diff
@@ -0,0 +1,93 @@
+#include <autoware_health_checker/diag_buffer.h>
+
+namespace autoware_health_checker
+{
+    DiagBuffer::DiagBuffer(std::string key, uint8_t type, std::string description, double buffer_length) : type(type), description(description)
+    {
+        key_ = key;
+        buffer_length_ = ros::Duration(buffer_length);
```

Comment:
What is the difference between the variable initialized at the same time as the init of the constructor and the variable initialized at a later time?

Reply:
It is const public variables.

Reply:
I hope the question is referring to why some are initialised that way and some are not. I have the same question... why is `key_` not initialised in the member initialiser list, for example?